swap_meet/vendor.py

```diff
@@ -1,2 +1,81 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):  #define class 
+        if not inventory:               #return empty list, if none
+            self.inventory = []         #attribute set-up
+
+        else:
+            self.inventory = inventory   #if inv data present
+        
+    def add(self, item):                  # a instance method
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item): 
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item    
+            
+        else:   
+            return False 
+            
+                #a instance method
+    def get_by_category(self, cat_name):
+        cat_list = []
+        for item in self.inventory:
+            if item.category == cat_name:
+                cat_list.append(item)
+        return  cat_list     
+        
+    def swap_items(self, friend, my_item, friend_item):
+        if my_item in self.inventory and friend_item in friend.inventory:
+            friend.inventory.append(my_item)
+            self.inventory.append(friend_item)
+            friend.inventory.remove(friend_item)
+            self.inventory.remove(my_item)
+            return True
+        return False
+
+    def swap_first_item(self, friend):
+        if not self.inventory or not friend.inventory:
+            return False
+        else:
+            
+            self.inventory.append(friend.inventory[0])
+            friend.inventory.append(self.inventory[0])
+
+            self.inventory.remove(self.inventory[0])
+            friend.inventory.remove(friend.inventory[0])
+            return True
+
+    def get_best_by_category(self, category):
+        if not self.inventory:
+            return None
+        else:
+            items = self.get_by_category(category)
+            if items: return max( items, key=lambda item: item.condition )
```

Comment:
Nice use of max with a key function to help find the item with the "best" condition. We've seen enough python now, that you could try writing your own version of `max`, including an optional `key` function parameter. Give it a shot!

Also, since `get_by_category` returns an empty list if it can't find any items of the desired category, we can do away with the initial empty list check. However, we _do_ need to to either check that thee items result isn't empty before using it with max, or else max will raise an error (or we could handle the error!).

From a style perspective, avoid squeezing `if` statements and their condition onto a single line. Also, make sure that both branches of the if end up returning something if one of them does. The existing code was correctly returning `None` as expected when nothing could be found, but this is because of the default Python behavior of returning None when there is no return statement. When one branch is returning something, it looks like an oversight to not return from the other branch, so make sure to explicitly return `None` if needed (here, rearranged under a guard clause).

```python
    def get_best_by_category(self, category):
        items = self.get_by_category(category)
        if not items:
            return None

        return max( items, key=lambda item: item.condition )
```